org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/A4Reporter.java

```diff
@@ -19,6 +19,8 @@
  * This class receives diagnostic, progress, and warning messages from Alloy4.
  * (This default implementation ignores all calls; you should subclass it to do
  * the appropriate screen output)
+ *
+ * @modified Nuno Macedo // [HASLab] electrum-temporal, electrum-decomposed
```

Comment:
two main changes to the Alloy reporter: the selected decomposed strategy is reported when the translation starts, and the solving process now reports in which step of the temporal analysis is.

temporal analysis is incremental, solving the command from 1 to the maximum number of steps defined in the scope, and reports whenever each of these steps finishes; the `SimpleReporter` overrides these messages in the log panel and shows only the most recent one.

Reply:
All these comments are very good, and it would make a lot of sense to move most of them to the @modified annotation